Update Revues.org to OpenEdition Journals

jfriviere opened the pull request:
Revues.org Platform has changed its name to OpenEdition Journals.
Journals URL have changed form http://XXX.revues.org to https://journals.openedition.org/XXX

I updated the translator Revues.org.js as follow.  Test cases are successfull.
Thanks !
JFR

Translator updates :
- Rename file from Revues.org.js to OpenEdition Journals.js
- New urls processing
- Abstract and tags retrieved by the translator are now abstract and tags in document language (were previously in the browsing language used by the user)
- Update urls in translator and test cases
- Update test cases

jfriviere commented:
I have fixed errors and warning reported by Travis.

jfriviere commented:
Hi all, 
A kind reminder for this pull request. 
Is there anything I can do to complete the request ?
Thanks

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -148,32 +163,36 @@ function doWeb(doc, url) {
 
-			ZU.processDocuments(urls, function(doc) {
-				scrape(doc, doc.location.href)
+			ZU.processDocuments(urls, function (doc) {
```
this should just work as `ZU.processDocuments(urls, scrape)` does it not?

jfriviere replied:
Yes, it does

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -131,13 +147,12 @@ function scrape(doc, url) {
 		}
 
-/* From old code: When is this needed?
+		/* From old code: When is this needed?
```
delete this block then

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
+			// The site lists all editor of journals as editor in the header. Turn them into contributors.
+			// I don't think there is a use case for editors for journal articles
+			if (item.itemType === "journalArticle") {
+				for (let i in item.creators) {
```
don't use `for ... in` to iterate through arrays. `for ... of` is preferred, explicitly increasing the index as you do elsewhere is also fine.

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
 function scrape(doc, url) {
-	//is this still necessary??
-	if (url.match(/persee\-\d+/)) {
+	var translator;
```
This is better handled by using `let translator = ...`  in each individual condition for clarity

jfriviere replied:
It's done

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
-		url.match(/document\d+/)) {
+	}
+	else if (ZU.xpath(doc, '//h1[@id="docTitle"]/span[@class="text"]').length
+		|| url.match(/document\d+/)) {
```
avoid using `.match()` for testing regex. use either `/regex/.test(string)` or `string.search(regex) != -1`

jfriviere replied:
it's fixed

adam3smith commented on OpenEdition Journals.js:
```diff
@@ -28,96 +48,92 @@ function detectWeb(doc, url) {
-	if (url.match(/persee\-\d+/)) {
+	var translator;
+	// Example: https://journals.openedition.org/remi/persee-144614
+	if (url.match(/persee-\d+/)) {
```
same as above on `match()`

jfriviere commented:
Thanks a lot for your review!
I've fixed all your comments and I rebased and squashed all the ESlint fixes in one commit. I hope I did the right way (not sure...)

jfriviere commented:
Hi @adam3smith 
I have fixed all the comments you made in your review on July. Is there anything else I should do? If not, could you merge the pull request so that Zotero will work back on http://journals.openedition.org? 
Thanks for your valuable work!
Jean-François

AbeJellinek commented:
Sorry this took so long to get merged!

jfriviere commented:
Thanks for merging! It's still relevant!
